pallets/scheduler/src/lib.rs

```diff
@@ -1354,21 +1211,23 @@ impl<T: Config> Pallet<T> {
                     } else {
                         task.maybe_periodic = None;
                     }
-                    let wake = now.saturating_add(period);
-                    match Self::place_task(wake, task) {
-                        Ok(new_address) => {
-                            if let Some(retry_config) = maybe_retry_config {
-                                Retries::<T>::insert(new_address, retry_config);
+                    let wake = now.saturating_add(&period);
+                    if let Some(wake) = wake {
+                        match Self::place_task(wake, task) {
+                            Ok(new_address) => {
+                                if let Some(retry_config) = maybe_retry_config {
+                                    Retries::<T>::insert(new_address, retry_config);
+                                }
+                            }
+                            Err((_, task)) => {
+                                // TODO: Leave task in storage somewhere for it to be rescheduled
```

Comment:
This TODO came from the scheduler pallet...can we get clarity on when this path could happen?